e2e-flap.sh

```
@@ -0,0 +1,789 @@
+echo "    Local advertisement state transitions during the window:"
+dump_sampler_transitions "$SVC_A"
+
+if [ "$reachable_now" = "no" ]; then
+    metric_valid="yes"
+    repro "EXPM/EXP4: $SVC_A stayed unreachable to the whole tailnet for ${SELFHEAL_WAIT}s after losing only its advertisement. Its serve config is present and correct, so DockTail - which reads current state from 'tailscale serve status', a view with no advertisement state in it - reported clean reconciles throughout and never repaired it."
+    diag "EXPM/EXP4 not self-healed"
+
+    step "Confirming a single 'tailscale serve advertise' is all it takes"
+    ts serve advertise "$SVC_A" >/dev/null 2>&1
+    back=$(wait_for_reachable "$SVC_A" 60)
+    [ "$back" = "-1" ] && harness "$SVC_A did not recover even after an explicit 'serve advertise'" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)